### DISCUSSION TITLE
How to use private tracker mode?

### DISCUSSION THREAD

**Discussion by Zerowalker** [torrust/torrust-tracker#476] [Q&A]
I Can't seem to figure it out.

I have set it to private, then generated a key, and then try to use the tracker: http://foo:1234/KEY/announce
But that doesn't seem to be what it expects, but that's normally how private tracker urls look like from what I gather.

**Comment by josecelano** [comment]
Hi @Zerowalker, the Torrust Tracker does not use that URL format, the KEY goes after the `announce` segment, like the following:

Announce request:

```console
curl "http://0.0.0.0:7070/announce/U7K9EiczJp5uA8kUavaSMd63Jp4OaSxQ?info_hash=%81%00%00%00%00%00%00%00%00%00%00%00%00%00%00%00%00%00%00%00&peer_addr=2.137.87.41&downloaded=0&uploaded=0&peer_id=-qB00000000000000001&port=17548&left=0&event=completed&compact=0"
```

Docs: https://docs.rs/torrust-tracker/3.0.0-alpha.11/torrust_tracker/servers/apis/v1/context/auth_key/index.html

Create key:

```console
$ curl -X POST "http://127.0.0.1:1212/api/v1/key/120?token=MyAccessToken"
{"key":"U7K9EiczJp5uA8kUavaSMd63Jp4OaSxQ","valid_until":1696846407,"expiry_time":"2023-10-09 10:13:27.791694159 UTC"}
```

- How did you generate the key? How are you using the tracker? With a client? Which one?
- Did you enable the HTTP tracker in the config file?

I think some trackers use a `key` GET param or a `passkey` GET param or the way you described `http://tracker:port/KEY/announce`. There is no BEP for this. Maybe we should support all options @da2ce7 ?

**Reply by Zerowalker** [reply]
Oh that explains it, thanks:)

**Reply by josecelano** [reply]
I've posted [here](https://www.reddit.com/r/BitTorrentProtocols/comments/174hurw/http_announce_request_url_format_for_private/?utm_source=share&utm_medium=web2x&context=3) to find out if there is one correct version or to know all the formats in case they are all valid.

**Reply by josecelano** [reply]
And also on [Stackoverflow](https://stackoverflow.com/q/77266333/3012842)

**Reply by extremlymtorrents** [reply]
Neeed to.encrypt the key or change it any ideea how

**Reply by josecelano** [reply]
> Neeed to.encrypt the key or change it any ideea how
> 

Hi @extremlymtorrents now you can only generate new keys using the API endpoint. I have to find it but I think there was a proposal or issue to upload your own keys. However, I think we have to keep the same length. I will check it.

**Reply by extremlymtorrents** [reply]
Send u an email on gmail
